omod/src/main/webapp/resources/htmlforms/section-plan.xml

```diff
@@ -123,45 +123,54 @@
 
             });
 
-            jq(function() {
+            var numMeds = 1;
+            var meds = [];
+            jq(function () {
 
-                var hasValue = function(element) {
+                var hasValue = function (element) {
                     return jq(element).find('.medication-name input').val();
                 };
 
-                var hideOtherMeds = function() {
-                    jq('#medication-2').hide();
-                    jq('#medication-3').hide();
-                    jq('#medication-4').hide();
-                    jq('#medication-5').hide();
-                    jq('#medication-6').hide();
-                    jq('#medication-7').hide();
-                    jq('#medication-8').hide();
+                var initMeds = function () {
+                    for (var i = 1; i &lt;= 8; i++) {
```

Comment:
Wait, so this is much nicer because it doesn't require hard-coding all the hides, etc, for each field, but does it still limit the number of medications to 8?  this was just an arbitrary restriction if we are now able to remove it.

We we also need to make sure we do some good testing once this is merged.

Reply:
Oh yeah! Lemme bump that number to something more unlikely to ever be hit. Like 30? I'll factor it out into a variable, too.